### PR TITLE
Add rz_vector_assign_at()

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Adds a `rz_vector_assign_at()` function.

**Test plan**

Added.

**Closing issues**

...

### CODE REVIEW

File: librz/util/vector.c [hunk 1]
```diff
@@ -471,6 +471,26 @@ RZ_API bool rz_pvector_join(RZ_NONNULL RzPVector *pvec1, RZ_NONNULL RzPVector *p
...
+/**
+ * \brief Assign the pointer \p ptr at \p index in the pvector.
+ *
+ * \param vec The pvector to assign to.
+ * \param index The index to assign the pointer to.
+ * \param ptr The pointer to assign.
+ *
+ * \return The pointer stored at \p index before. Or NULL in case of failure.
+ */
+RZ_API void *rz_pvector_assign_at(RZ_BORROW RZ_NONNULL RzPVector *vec, size_t index, RZ_OWN RZ_NONNULL void *ptr) {
```
Comment (wargio) [inline librz/util/vector.c:483]: why you need this when `rz_pvector_set` is available?

**Review by wargio — CHANGES_REQUESTED** [review]
.

File: librz/util/vector.c (hunk 1 above)
Comment (Rot127) [reply librz/util/vector.c:483]: `rz_pvector_set` is private. My thinking was, that it is better to have an equivalent function to the normal `rz_vector_assign_at`. So we don't mix up names with function between `RzVector` and `RzPVector`.

Comment (wargio) [reply librz/util/vector.c:483]: well, it's an inline within the headers.

Comment (Rot127) [reply librz/util/vector.c:483]: Something is wrong with this week -.-
Yeah, it is not necessary to add it than. `rz_pvector_get` and `rz_pvector_set` is good enough. Let me close this one.

**Pull request closed without merging**